### PR TITLE
Components: Move SearchListControl to woocommerce/components

### LINKED ISSUE #417: Move SearchListControl to woocommerce/components
Other people should be able to use this base component. Let's move it to the @woocommerce/components npm package. This is a tracking issue to create a pr into [wc-admin.](https://github.com/woocommerce/wc-admin)

It will need…

- Documentation
- A DevDocs example
- Another a11y review
- Check in IE11

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #417 – SearchListControl is a useful component for extension blocks, so let's move it to the wc.components library. There's a corresponding PR in wc-admin to add these files. There should be no visible or functional change here.

### How to test the changes in this Pull Request:

1. Follow the steps in the other PR https://github.com/woocommerce/wc-admin/pull/1628

### CODE REVIEW

**Review by timmyc — APPROVED** [review]
Tested in conjunction with the wc-admin PR. All looking good here as well!

**Comment by ryelle** [comment]
Merging this will need to wait for a new release of @woocommerce/components, once the PR over there is merged

**Pull request merged**
